src/permissions/rules/rules.service.ts

```diff
@@ -26,20 +28,40 @@ export class RulesService {
     const permissionDbName = this.configService.get(
       RulesService.ENV_PERMISSION_DB,
     );
-    this.loadRules(permissionDbName);
+    this.loadRulesContinuously(permissionDbName).subscribe();
   }
 
-  async loadRules(db: string): Promise<RulesConfig> {
-    this.permission = undefined;
-    return firstValueFrom(
-      this.couchdbService.get<Permission>(db, Permission.DOC_ID).pipe(
-        retry({ count: 60, delay: 1000 }),
-        map((doc) => doc.data),
-        tap((permission) => (this.permission = permission)),
+  loadRulesContinuously(db = 'app') {
+    // Rebuild params when observable is retried/repeated
+    const getParams = defer(() =>
+      of({
+        filter: '_doc_ids',
+        feed: 'longpoll',
+        limit: '1',
+        since: this.lastSeq,
+        include_docs: true,
+        doc_ids: JSON.stringify([Permission.DOC_ID]),
+      }),
+    );
+
+    return getParams.pipe(
+      concatMap((params) =>
+        this.couchdbService.get<ChangesResponse>(db, '_changes', params),
       ),
+      tap((changes) => {
+        this.lastSeq = changes.last_seq;
+        if (changes.results.length > 0) {
+          this.permission = changes.results[0].doc.data;
+        }
+      }),
+      catchError((err) => {
+        console.error('LOAD RULES ERROR:', err);
+        throw err;
+      }),
+      retry({ delay: 1000 }),
+      repeat(),
     );
```

Comment:
any special reason why to return an observable here that doesn't return things to use in subscribe but in itself repeats once it is subscribed?

Wouldn't it be less convoluted to subscribe to a repeating couchdb request here, move the actual action from `tap` to `subscribe` and not return anything?

Reply:
Good point. Initial had only allowed a maximum amount of errors but this will now repeat forever so the observable doesn't help much.